app/forms/publish_interface/course_vacancies_form.rb

```diff
@@ -0,0 +1,106 @@
+module PublishInterface
+  class CourseVacanciesForm < BaseModelForm
+    alias_method :course, :model
+
+    FIELDS = %i[
+      site_statuses
+      change_vacancies_confirmation
+      has_vacancies
+    ].freeze
+
+    attr_accessor(*FIELDS)
+
+    delegate :recruitment_cycle_year, :provider_code, :running_site_statuses, :name, to: :course
+
+    validate :vacancies_confirmation
+
+    def initialize(*params)
+      @site_statuses = []
+      super
+    end
+
+    def site_statuses_attributes=(attributes)
+      @site_statuses = attributes.values
+    end
+
+  private
+
+    def compute_fields
+      course
+        .attributes
+        .symbolize_keys
+        .slice(*FIELDS)
+        .merge(new_attributes)
+        .merge(has_vacancies: course.has_vacancies?)
+    end
+
+    def assign_attributes_to_model
+      model.assign_attributes(
+        fields
+          .symbolize_keys
+          .except(*fields_to_ignore_before_save)
+          .deep_merge(attributes_for_site_statuses),
+      )
```

Comment:
It looks a little be wieldy, can we change it to something like instead please
```ruby
      model.assign_attributes(model_attributes)

...

      def model_attributes
        fields
          .symbolize_keys
          .except(*fields_to_ignore_before_save)
          .deep_merge(attributes_for_site_statuses)
      end
```

Reply:
agreed, updated